plugins/carla/carla.c

```diff
@@ -0,0 +1,487 @@
+/*
+ * Carla plugin for OBS
+ * Copyright (C) 2023 Filipe Coelho <falktx@falktx.com>
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+#include <obs-module.h>
+#include <util/platform.h>
+
+#ifndef CARLA_MODULE_ID
```

Comment:
Are these macros required? Only use macros if there is no other language feature to achieve the same (e.g. const character array for constant strings).

Reply:
The macros allow to reuse the case code for other OBS plugins, which I already have written for testing but are not part of this PR. The idea is to build a few other variants that reuse the same base, but are separate OBS plugins.
One of them is "carla patchbay" which brings the full GUI along as seen in https://user-images.githubusercontent.com/1334853/235544415-6ae2c9e5-3cbf-47b2-8fc0-013f0ec8276d.png

I also dont know if calling it "Carla" is okay or not, you might prefer something more like "OBS audio plugin".

Reply:
> The macros allow to reuse the same code for other OBS plugins, which I already have written for testing but are not part of this PR. The idea is to build a few other variants that reuse the same base, but are separate OBS plugins. One of them is "carla patchbay" which brings the full GUI along as seen in https://user-images.githubusercontent.com/1334853/235544415-6ae2c9e5-3cbf-47b2-8fc0-013f0ec8276d.png

Understandable, but this code is targeted to become part of the obs-studio codebase and should be designed for that. If this were a 3rd party "audio plugin" template, then this would make sense, but for a core plugin this configurability is not needed and should be removed.

> I also dont know if calling it "Carla" is okay or not, you might prefer something more like "OBS audio plugin".

I don't mind keeping the name of the technology in place, maybe others have an opinion on this.

Reply:
note that I already pushed the 2 OBS modules in this PR, so what I said above regarding carla-bridge and carla-patchbay reusing the same code already applies.
the same code is built twice, but uses a compiler macro to tweak it just enough to be different. the alternative is to have duplicate files where they only differ for 2-3 simple strings. is that better than the current approach?

Reply:
I have to admit that the entire architecture of this plugin goes above my head right now because there's multiple binaries, libraries, modules - it goes beyond the complexity of any other plugin we have right now.

Do you have an overview of all these moving parts somewhere and what the differences between them are? Because I don't currently understand why we need 2 plugins with the same code and what their differences would be.

Reply:
Thank you for the write-up, will read this carefully later tonight or tomorrow.